src/main/java/be/seeseemelk/mockbukkit/block/BlockMock.java

```diff
@@ -98,6 +98,25 @@ public BlockMock(@NotNull Material material, @Nullable Location location)
 		this.blockData = BlockDataMock.mock(material);
 	}
 
+
+	/**
+	 * Creates an instance of BlockMock if it isn't already mocked
+	 *
+	 * @param block block to be mocked
+	 * @return mocked block
+	 */
+	public static BlockMock mock(Block block)
+	{
+		if (block instanceof BlockMock blockMock)
+		{
+			return blockMock;
+		}
+		else
+		{
+			return new BlockMock(block.getLocation());
+		}
+	}
```

Comment:
I think throwing an exception here is more fit, if Block is not an BlockMock instance, then there's already a huge issue. I would really just do this (not in it's separate method):
```java
BlockMock blockMock = (BlockMock) block;
```